src/structures/ApplicationCommand.js

```diff
@@ -214,7 +214,7 @@ class ApplicationCommand extends Base {
       type: typeof option.type === 'number' && !received ? option.type : ApplicationCommandOptionTypes[option.type],
       name: option.name,
       description: option.description,
-      required: option.required,
+      required: option.required ?? false,
```

Comment:
I'm not sure this is needed, because once you stringify this the required key will be dropped, so it would correct as require by discord be optional and not be sent in the payload.

Reply:
But this data isn’t sent anywhere, it’s the data that you get when you fetch the commands from Discord. If that’s not what you mean then I’m not sure I understand

Reply:
Right, I was under the assumption the transform is also used before stringifying it and sending it to discord 👍

Reply:
And it is which is why you have the received boolean but that only affects the type value